hilti/toolchain/src/compiler/driver.cc

```diff
@@ -431,14 +431,19 @@ Result<Nothing> Driver::parseOptions(int argc, char** argv) {
 
             case 'h': usage(); return Nothing();
 
-            case '?': usage(); return error("unknown option");
+            case '?':
+                std::cerr << "Try '" << argv[0] << " --help' for usage\n";
+                if ( optopt )
+                    fatalError(fmt("option '%s' requires an argument", argv[optind - 1]));
+                else
+                    fatalError(fmt("option '%s' not supported", argv[optind - 1]));
```

Comment:
How about making this just one line of output:

```suggestion
            case '?':
                if ( optopt )
                    fatalError(fmt("option '%s' requires an argument; try --help for usage", argv[optind - 1]));
                else
                    fatalError(fmt("option '%s' not supported; try --help for usage", argv[optind - 1]));
```